### PR TITLE
25330 publish WIP

### PR DESCRIPTION
**Pull request by Angamanga**
Adding step4, "publish" to targeted survey-wizard
NOTE: This is not ready to merge, needs some work for the notifications etc
Have not added any tests until we know the details for the api

Test-checklist in issue: https://app.zenhub.com/workspace/o/ushahidi/platform/issues/2530 
Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

File: app/settings/surveys/targeted-surveys/targeted-survey-edit.html [hunk 1]
```diff
@@ -180,9 +188,28 @@ <h2 class="stepper-heading">
                                 <span class="stepper-badge">4</span>
                                 <span translate="survey.targeted_survey.publish"></span>
                             </h2>
+                            <div class="alert" ng-show="isActiveStep(4)">
+                                <p translate="survey.targeted_survey.publish_description" translate-values="{questions: targetedSurvey.stepThree.questions.length, numbers:finalNumbers.goodNumbers.length, sms: sms, cost:cost}"></p>
+                            </div>
+                            <div class="form-field switch checked">
```
Comment (crcommons) [inline app/settings/surveys/targeted-surveys/targeted-survey-edit.html]: These should be hidden until you're in step 4

File: app/settings/surveys/targeted-surveys/targeted-survey-edit.html [hunk 2]
```diff
@@ -180,9 +188,28 @@ <h2 class="stepper-heading">
...
+                                <label for="switch4" translate="survey.require_review"></label>
+                                    <p ng-show="editQuestion.require_review" translate="survey.require_review_desc"></p>
+                                    <div class="toggle-switch">
+                                        <input class="tgl" id="switch4" type="checkbox" ng-model="targetedSurvey.stepFour.require_review">
+                                        <label class="tgl-btn" for="switch4"></label>
+                                    </div>
+                            </div>
+                             <div class="form-field switch checked">
```
Comment (crcommons) [inline app/settings/surveys/targeted-surveys/targeted-survey-edit.html]: This should also be hidden until you get to step 4

File: app/settings/surveys/targeted-surveys/targeted-survey-edit.html [hunk 3]
```diff
@@ -180,9 +188,28 @@ <h2 class="stepper-heading">
                                 <span class="stepper-badge">4</span>
                                 <span translate="survey.targeted_survey.publish"></span>
                             </h2>
+                            <div class="alert" ng-show="isActiveStep(4)">
+                                <p translate="survey.targeted_survey.publish_description" translate-values="{questions: targetedSurvey.stepThree.questions.length, numbers:finalNumbers.goodNumbers.length, sms: sms, cost:cost}"></p>
```
Comment (crcommons) [inline app/settings/surveys/targeted-surveys/targeted-survey-edit.html]: So technically there should be 4 possible text options:
- option 1: 1 questions to 1 person
- option 2: 1 questions to 2+ people
- option 3: 2+ questions to 1 person
- option 4: 2+ questions to 2+ people

I don't know any other way around this to make sure that the translations are correct.

Comment (crcommons) [reply app/settings/surveys/targeted-surveys/targeted-survey-edit.html]: You can see similar issues in my latest PR where I had to put in multiple translation options. https://github.com/ushahidi/platform-client/pull/1036

File: app/settings/surveys/targeted-surveys/targeted-edit.controller.js [hunk 4]
```diff
@@ -205,8 +240,63 @@ function (
...
+        let survey =  {
+                color: null,
+                everyone_can_create: true,
+                name: $scope.targetedSurvey.stepOne.name.$modelValue,
+                description: $scope.targetedSurvey.stepOne.description.$modelValue,
+                require_approval: $scope.targetedSurvey.stepFour.require_review,
+                hide_author: $scope.targetedSurvey.stepFour.hide_responders
+            };
+
+        Notify.confirmModal('Are you sure you want to send this SMS survey?', null, 'We will send {{questions}} questions to {{numbers}} people via SMS for a total of {{sms}} messages. This could cost up to USD.', 'publish').then(function () {
```
Comment (crcommons) [inline app/settings/surveys/targeted-surveys/targeted-edit.controller.js]: These variables aren't being interpolated, not sure why.

File: app/settings/surveys/targeted-surveys/targeted-survey-edit.html (hunk 1 above)
Comment (Angamanga) [reply app/settings/surveys/targeted-surveys/targeted-survey-edit.html]: fixed!

File: app/settings/surveys/targeted-surveys/targeted-survey-edit.html (hunk 3 above)
Comment (Angamanga) [reply app/settings/surveys/targeted-surveys/targeted-survey-edit.html]: Good catch, fixed now!

File: app/settings/surveys/targeted-surveys/targeted-survey-edit.html (hunk 2 above)
Comment (Angamanga) [reply app/settings/surveys/targeted-surveys/targeted-survey-edit.html]: Fixed

**Pull request merged**